iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/twin/DesiredPropertiesTests.java

```diff
@@ -335,4 +344,116 @@ public void testUpdateDesiredProperties() throws IOException, InterruptedExcepti
 
         removeMultipleDevices(MAX_DEVICES);
     }
+
+    @AllArgsConstructor
+    class TwinPropertiesCallbackImpl implements TwinPropertiesCallback
+    {
+        TwinCollection expectedProperties;
+
+        @Override
+        public void TwinPropertiesCallBack(TwinCollection actualProperties, Object context) {
+            Success desiredPropertiesCallbackState = (Success) context;
+            desiredPropertiesCallbackState.callbackWasFired();
+
+            if (actualProperties.size() != expectedProperties.size())
+            {
+                log.error("Batch twin properties callback was fired, but had an unexpected number of properties");
+                desiredPropertiesCallbackState.setResult(false);
+                return;
+            }
+
+            for (String actualPropertyKey : actualProperties.keySet())
+            {
+                if (!expectedProperties.containsKey(actualPropertyKey))
+                {
+                    log.error("Batch twin properties callback was fired, but had an unexpected key");
+                    desiredPropertiesCallbackState.setResult(false);
+                    return;
+                }
+                else if (!actualProperties.get(actualPropertyKey).equals(expectedProperties.get(actualPropertyKey)))
+                {
+                    log.error("Batch twin properties callback was fired, but had an unexpected value");
+                    desiredPropertiesCallbackState.setResult(false);
+                    return;
+                }
+            }
+
+            // If the twins have the same number of properties, and the same key/value pairs, then the expected
+            // twin matches the actually received twin
+            desiredPropertiesCallbackState.setResult(true);
+        }
+    }
+
+    // This test is for the startDeviceTwin/startTwin API that takes the TwinPropertiesCallback rather than the TwinPropertyCallback
+    // This callback should receive the full twin update in one callback, rather than one callback per updated
+    // desired property
+    @Test
+    @StandardTierHubOnlyTest
+    public void testSubscribeToDesiredPropertiesBatch() throws Exception
```

Comment:
This test runs with parameters for protocols (AMQPS/MQTT/AMQPS_WS/MQTT_WS), for authentication types (SAS/x509), and for a few other configurable options at the client level, for what that's worth.